chain/rust/src/transaction/utils.rs

```diff
@@ -17,48 +17,49 @@ impl TransactionOutput {
         script_reference: Option<ScriptRef>,
     ) -> Self {
         match (datum_option, script_reference) {
-            (None, None) => Self::ShelleyTxOut(ShelleyTxOut::new(address, amount)),
+            (None, None) => Self::AlonzoFormatTxOut(AlonzoFormatTxOut::new(address, amount)),
             (Some(DatumOption::Hash { datum_hash, .. }), None) => {
-                Self::AlonzoTxOut(AlonzoTxOut::new(address, amount, datum_hash))
+                let mut tx_out = AlonzoFormatTxOut::new(address, amount);
+                tx_out.datum_hash = Some(datum_hash);
+                Self::AlonzoFormatTxOut(tx_out)
             }
             (datum, script_ref) => {
-                let mut tx_out = BabbageTxOut::new(address, amount);
+                let mut tx_out = ConwayFormatTxOut::new(address, amount);
                 tx_out.datum_option = datum;
                 tx_out.script_reference = script_ref;
-                Self::BabbageTxOut(tx_out)
+                Self::ConwayFormatTxOut(tx_out)
             }
         }
     }
 
     pub fn address(&self) -> &Address {
         match self {
-            Self::ShelleyTxOut(tx_out) => &tx_out.address,
-            Self::AlonzoTxOut(tx_out) => &tx_out.address,
-            Self::BabbageTxOut(tx_out) => &tx_out.address,
+            Self::AlonzoFormatTxOut(tx_out) => &tx_out.address,
+            Self::ConwayFormatTxOut(tx_out) => &tx_out.address,
         }
     }
 
     pub fn amount(&self) -> &Value {
         match self {
-            Self::ShelleyTxOut(tx_out) => &tx_out.amount,
-            Self::AlonzoTxOut(tx_out) => &tx_out.amount,
-            Self::BabbageTxOut(tx_out) => &tx_out.amount,
+            Self::AlonzoFormatTxOut(tx_out) => &tx_out.amount,
+            Self::ConwayFormatTxOut(tx_out) => &tx_out.amount,
```

Comment:
babbage / shelley are subset of conway right?

Reply:
CDDL:
```cddl
alonzo_format_tx_out = [
  address,
  amount : value,
  ? datum_hash
]
conway_format_tx_out = {
    0 : address,      ; @name address
    1 : value,        ; @name amount
  ? 2 : datum_option, ; @name datum_option
  ? 3 : script_ref,   ; @name script_reference
}

transaction_output = alonzo_format_tx_out / conway_format_tx_out
```

Starting from Alonzo they added the optional datum hash to the format (when left out it's the same as Shelley), and in Babbage they changed it completely changed it to be a map but kept the previous array format as well as it takes up less space on-chain (no need to store keys) + maybe some compatability concerns.

The reason I changed this for Conway is that the Babbage one contained a Babbage-era script ref (plutus v1/v2) whereas this one is a Conway-era one (plutus v1/v2/v3).